src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -8,393 +8,131 @@
 #include <shogun/mathematics/Math.h>
 #include <shogun/mathematics/linalg/LinalgNamespace.h>
-#include <shogun/mathematics/RandomNamespace.h>
+#include <shogun/mathematics/NormalDistribution.h>
 #include <vector>
 #include <algorithm>
 
 using namespace shogun;
 
-void RandomFourierGaussPreproc::copy(const RandomFourierGaussPreproc & feats) {
-
-	dim_input_space = feats.dim_input_space;
-	cur_dim_input_space = feats.cur_dim_input_space;
-
-	dim_feature_space = feats.dim_feature_space;
-	cur_dim_feature_space=feats.cur_dim_feature_space;
-
-	kernelwidth=feats.kernelwidth;
-	cur_kernelwidth=feats.cur_kernelwidth;
-
-	if(cur_dim_feature_space>0)
-	{
-		if(feats.randomcoeff_additive==NULL)
-		{
-			throw ShogunException(
-							"void CRandomFourierGaussPreproc::copy(...): feats.randomcoeff_additive==NULL && cur_dim_feature_space>0 \n");
-		}
-
-		randomcoeff_additive = SG_MALLOC(float64_t, cur_dim_feature_space);
-		std::copy(feats.randomcoeff_additive,feats.randomcoeff_additive+cur_dim_feature_space,randomcoeff_additive);
-	}
-	else
-	{
-		randomcoeff_additive = NULL;
-	}
-
-	if((cur_dim_feature_space>0)&&(cur_dim_input_space>0))
-	{
-		if(feats.randomcoeff_multiplicative==NULL)
-		{
-			throw ShogunException(
-							"void CRandomFourierGaussPreproc::copy(...): feats.randomcoeff_multiplicative==NULL && cur_dim_feature_space>0 &&(cur_dim_input_space>0)  \n");
-		}
-
-		randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-		std::copy(feats.randomcoeff_multiplicative,feats.randomcoeff_multiplicative+cur_dim_feature_space*cur_dim_input_space,randomcoeff_multiplicative);
-	}
-	else
-	{
-		randomcoeff_multiplicative = NULL;
-	}
-
-}
-
-RandomFourierGaussPreproc::RandomFourierGaussPreproc() :
-	RandomMixin<DensePreprocessor<float64_t>> () {
-	dim_feature_space = 1000;
-	dim_input_space = 0;
-	cur_dim_input_space = 0;
-	cur_dim_feature_space=0;
-
-	randomcoeff_multiplicative=NULL;
-	randomcoeff_additive=NULL;
-
-	kernelwidth=1;
-	cur_kernelwidth=kernelwidth;
-
-	SG_ADD(&dim_input_space, "dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&cur_dim_input_space, "cur_dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&dim_feature_space, "dim_feature_space",
-			"Dimensionality of the feature space.");
-	SG_ADD(&cur_dim_feature_space, "cur_dim_feature_space",
-			"Dimensionality of the feature space.");
-
-	SG_ADD(&kernelwidth, "kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-	SG_ADD(&cur_kernelwidth, "cur_kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-
-	watch_param(
-		"randomcoeff_additive", &randomcoeff_additive,
-		&cur_dim_feature_space);
-
-	watch_param(
-		"randomcoeff_multiplicative", &randomcoeff_multiplicative,
-		&cur_dim_feature_space, &cur_dim_input_space);
+RandomFourierGaussPreproc::RandomFourierGaussPreproc()
+    : RandomMixin<DensePreprocessor<float64_t>>()
+{	
+	m_fitted = false;
```

Comment:
I would prefer to not have this member, but instead infer whether it is fitted from the presence of the basis